kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/package-info.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+/**
+ *
+ * <h2 id='implementing'>Implementing Filters</h2>
+ *
```

Comment:
We should probably have some text here about:

- Thread safety
- The fact that filter instances are scoped to a specific client connection
- Maybe something about sharing state?

 Potentially could be left to an alternate PR.

Reply:
Raised as separate issue (#557)